exercise/models.py

```diff
@@ -118,40 +127,38 @@ def __str__(self):
     length_minutes = models.IntegerField()
     demographics = models.ManyToManyField(DemographicsInfo, blank=True)
     emails = models.ManyToManyField(ExerciseEmail, blank=True)
-    email_reveal_times = PickledObjectField(null=True)
 
     def set_email_reveal_times(self):
-        # generate email reveal times - these are unique per exercise and are stored in seconds
-        # TODO: (TCKT-1235) minimum amount of emails that should have reveal_time 0
-        # in case the 10% of emails is less than 1 email
-        reveal_times = []
-
-        emails = self.emails.all()
-        if not emails:
+        if not self.emails.all():
             return
+        # Set some emails based on a threshold to zero time initially.
+        threshold_emails = int(ceil((self.emails.count() * settings.REVEAL_TIME_ZERO_THRESHOLD)))
 
-        for email in emails:
-            reveal_times.append(
-                {
-                    'email_id': email.id,
-                    'reveal_time':  randint(0, self.length_minutes * 60)
-                }
-            )
+        while threshold_emails > 0:
+            email = self.emails.all()[randint(0, self.emails.count()-1)]
```

Comment:
I know the chances are very small but you can potentially end up picking the same item if you keep on calling `self.emails.all()`

I would corner back to the logic which was there before:

     # at the top of the function
     emails = self.emails.all()
        if not emails:	
            return

Then you could just 

    while received_emails_count:
        email = emails.pop(random.randrange(len(emails)))
        # Use filter in case ExerciseEmailProperties cannot be found
        email_properties = ExerciseEmailProperties.objects.filter(exercise_id=self.id, email_id=email.id).first()
        if email_properties:
            email_properties.set_reveal_time(0)
        received_emails_count -= 1

Reply:
Done

Reply:
But you cannot use pop against a QuerySet like that my original method works.